web/pages/groups.tsx

```diff
@@ -0,0 +1,206 @@
+import { getUser, User } from 'web/lib/firebase/users'
+import { Tabs } from 'web/components/layout/tabs'
+import { GroupMembersList } from 'web/pages/group/[...slugs]'
+import { checkAgainstQuery } from 'web/hooks/use-sort-and-query-params'
+
+export async function getStaticProps() {
+  const groups = await listAllGroups().catch((_) => [])
+
+  const creators = await Promise.all(
+    groups.map((group) => getUser(group.creatorId))
+  )
+  const creatorsDict = Object.fromEntries(
+    creators.map((creator) => [creator.id, creator])
+  )
+
+  return {
+    props: {
+      groups: groups,
+      creatorsDict,
+    },
+
+    revalidate: 60, // regenerate after a minute
+  }
+}
+
+export default function Groups(props: {
+  groups: Group[]
+  creatorsDict: { [k: string]: User }
+}) {
+  const [creatorsDict, setCreatorsDict] = useState(props.creatorsDict)
+
+  const groups = useGroups() ?? props.groups
+  const user = useUser()
+  const memberGroupIds = useMemberGroupIds(user) || []
+
+  useEffect(() => {
+    // Load User object for creator of new Groups.
+    const newGroups = groups.filter(({ creatorId }) => !creatorsDict[creatorId])
+    if (newGroups.length > 0) {
+      Promise.all(newGroups.map(({ creatorId }) => getUser(creatorId))).then(
+        (newUsers) => {
+          const newUsersDict = Object.fromEntries(
+            newUsers.map((user) => [user.id, user])
+          )
+          setCreatorsDict({ ...creatorsDict, ...newUsersDict })
+        }
+      )
+    }
+  }, [creatorsDict, groups])
+
+  const [query, setQuery] = useState('')
+
+  // List groups with the highest question count, then highest member count
+  // TODO use find-active-contracts to sort by?
+  const matches = sortBy(groups, [
+    (group) => -1 * group.contractIds.length,
+    (group) => -1 * group.memberIds.length,
+  ]).filter(
+    (g) =>
+      checkAgainstQuery(query, g.name) ||
+      checkAgainstQuery(query, g.about || '') ||
+      checkAgainstQuery(query, creatorsDict[g.creatorId].username)
+  )
+
+  const matchesOrderedByRecentActivity = sortBy(groups, [
+    (group) => -1 * group.mostRecentActivityTime,
+  ]).filter(
+    (g) =>
+      checkAgainstQuery(query, g.name) ||
+      checkAgainstQuery(query, g.about || '') ||
+      checkAgainstQuery(query, creatorsDict[g.creatorId].username)
+  )
+
+  // Not strictly necessary, but makes the "hold delete" experience less laggy
+  const debouncedQuery = debounce(setQuery, 50)
+
+  return (
+    <Page>
+      <Col className="items-center">
+        <Col className="w-full max-w-xl">
+          <Col className="px-4 sm:px-0">
+            <Row className="items-center justify-between">
+              <Title text="Explore groups" />
+              {user && (
+                <CreateGroupButton user={user} goToGroupOnSubmit={true} />
+              )}
+            </Row>
+
+            <div className="mb-6 text-gray-500">
+              Groups enable users to curate, discuss, and rank in a subset of
```

Comment:
Discuss and compete on questions with a group of friends.

Reply:
Nice!